app/views/modules/player/_video_js_element.html.erb

```diff
@@ -0,0 +1,235 @@
+    <video-js id="video-js-embed-<%= @master_file.id %>"
+            class="vjs-big-play-centered"
+            controls
+            style="width:100%; height:100%;"
+            data-setup='<%= @videojs_options %>'>
+      <% section_info[:stream_hls].each do |hls| %>
+        <source src="<%= hls[:url] %>" type="application/x-mpegURL" data-quality="<%= hls[:quality] %>" label="<%= hls[:quality] %>"/>
+      <% end %>
+      <% if section_info[:caption_paths].present? %>
+        <% section_info[:caption_paths].each do |c| %>
+          <track <% if c[:label] %>label="<%= c[:label] %>" <% end %> srclang="<%= c[:language] %>" kind="subtitles" type="<%= c[:mime_type] %>" src="<%= c[:path] %>"></track>
+        <% end %>
+      <% end %>
+    </video-js>
+    <% if section_info[:is_video] %>
+      <label class='video-title' id='video-title-<%= @master_file.id %>'>
+        <a class='video-link' id='video-link-<%= @master_file.id %>' href="<%= master_file_url(@master_file.id) %>" target="_blank" rel="noreferrer noopener"><%= @master_file.title || @master_file.media_object.title %></a>
+      </label>
+    <% end %>
+  </div>
+<% end %>
+
+<% # These script tags are for pulling in videojs and the quality selector plug-in from their CDNs.
+   # May be worth considering pulling them into our local asset pipeline. %>
+<script src="https://vjs.zencdn.net/7.10.2/video.min.js"></script>
+<script src="https://unpkg.com/@silvermine/videojs-quality-selector/dist/js/silvermine-videojs-quality-selector.min.js"></script>
+
+<% content_for :page_scripts do %>
+  <script>
+    var player = videojs(document.querySelector('#video-js-embed-<%= @master_file.id %>'));
+    var ButtonComponent = videojs.getComponent('Button');
+    var viewInRepoButton = new ButtonComponent(player, {
+      clickHandler: function(event) {
+        window.open('<%= master_file_url(@master_file.id) %>', '_blank').focus();
+      }
+    });
+
+    viewInRepoButton.addClass('vjs-custom-external-link');
+    let newButton = player.controlBar.addChild(viewInRepoButton, {}, <%= section_info[:is_video] ? 8 : 6 %>);
+    newButton.controlText('View in Repository');
+
+    document.querySelector('.vjs-custom-external-link .vjs-icon-placeholder').innerHTML = '<svg xmlns="http://www.w3.org/2000/svg" height="1em" viewBox="0 0 512 512"><!--! Font Awesome Free 6.4.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license (Commercial License) Copyright 2023 Fonticons, Inc. --><style>svg{fill:#ffffff}</style><path d="M320 0c-17.7 0-32 14.3-32 32s14.3 32 32 32h82.7L201.4 265.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L448 109.3V192c0 17.7 14.3 32 32 32s32-14.3 32-32V32c0-17.7-14.3-32-32-32H320zM80 32C35.8 32 0 67.8 0 112V432c0 44.2 35.8 80 80 80H400c44.2 0 80-35.8 80-80V320c0-17.7-14.3-32-32-32s-32 14.3-32 32V432c0 8.8-7.2 16-16 16H80c-8.8 0-16-7.2-16-16V112c0-8.8 7.2-16 16-16H192c17.7 0 32-14.3 32-32s-14.3-32-32-32H80z"/></svg>'
+
```

Comment:
We could try to use the class `vjs-custom-external-link` to set the SVG using `background-image` in CSS. An example implementation from Ramp: https://github.com/samvera-labs/ramp/blob/main/src/components/MediaPlayer/VideoJS/components/styles/VideoJSFileDownload.scss#L16. 
If it's hard to do with CSS scoping with videojs classes, the current implementation is good too :+1: 
What do you think?

Reply:
I tried to move it to CSS based on that Ramp example but was unable to get the icon to render, so I'm going to leave the implementation as is for now.